src/WebJobs.Extensions.DurableTask.Analyzers/SyntaxNodeUtils.cs

```diff
@@ -29,16 +29,24 @@ public static DurableVersion GetDurableVersion(SemanticModel semanticModel)
 
         public static bool TryGetSemanticModelForSyntaxTree(SemanticModel model, SyntaxNode node, out SemanticModel newModel)
         {
-            if (model?.SyntaxTree == null || node?.SyntaxTree == null)
+            try
+            {
+                if (model?.SyntaxTree == null || node?.SyntaxTree == null)
+                {
+                    newModel = null;
+                    return false;
+                }
+
+                newModel = model.SyntaxTree == node.SyntaxTree
+                    ? model
+                    : model.Compilation.GetSemanticModel(node.SyntaxTree);
+                return newModel != null;
+            }
+            catch (ArgumentException)
```

Comment:
ArgumentException will get thrown when the syntax tree is not in the compilation. Because I'm passing around SemanticModels I've seen this exception a few times. It breaks the analyzer. This exception kind of haunts me (also null reference exceptions).

I added this while testing but I realize now that I fixed the problem I had where I was using the "wrong semantic model" so I could take this out. In theory no one should get this exception.